### PR TITLE
Grpc client port default grpc server port

### LINKED ISSUE #19219: gRPC client ports should default to `quarkus.grpc.server.port`, not 9000
### Describe the bug

Checkout `https://github.com/quarkusio/quarkus-quickstarts`, then go to `grpc-plain-text-quickstart`and run `mvn quarkus:dev`.

If you browse to `http://localhost:8080/hello/blocking/QE`, you should get `Hello QE! HelloWorldService has been called 1 number of times.`. So good so far.

The problem is when I set a different gRPC server port:

`mvn -Dquarkus.grpc.server.port=3889 quarkus:dev`

The app will start as expected and the gRPC will listen at the expected port:

```
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2021-08-04 13:27:54,999 INFO  [io.qua.grp.run.GrpcServerRecorder] (Quarkus Main Thread) Registering gRPC reflection service
2021-08-04 13:27:55,084 INFO  [io.qua.grp.run.GrpcServerRecorder] (vert.x-eventloop-thread-0) gRPC Server started on 0.0.0.0:3889 [SSL enabled: false]
2021-08-04 13:27:55,136 INFO  [io.quarkus] (Quarkus Main Thread) grpc-plain-text-quickstart 1.0.0-SNAPSHOT on JVM (powered by Quarkus 2.1.0.Final) started in 1.957s. Listening on: http://localhost:8080
2021-08-04 13:27:55,137 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
2021-08-04 13:27:55,138 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [cdi, grpc-client, grpc-server, resteasy, resteasy-mutiny, smallrye-context-propagation, vertx]
```

However, browsing again to `http://localhost:8080/hello/blocking/QE` will fail as the client is still listening at `9000`. 

In order to workaround this, I had to provide an additional property:

`mvn -Dquarkus.grpc.server.port=3889 -Dquarkus.grpc.clients.hello.port=3889 quarkus:dev`

And then the URL will work. However, the second property should be unnecessary.

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Sgitario**
At the moment, when running a Quarkus application with a gRPC server and changing the default port other than 9000, we need to set both `quarkus.grpc.server.port` and `quarkus.grpc.clients.XXX.port` properties.

This PR fixes this issue by setting the clients property the `quarkus.grpc.server.port` value by default instead of 9000.

Fix https://github.com/quarkusio/quarkus/issues/19219

### CODE REVIEW

**Comment by michalszynkiewicz** [comment]
@Sgitario could you write a bit more on the motivation of this change?

A (configured) client will usually be used to talk to a service that is exposed by a different application, no?

**Comment by michalszynkiewicz** [comment]
@mkouba WDYT?

**Comment by mkouba** [comment]
I'm not sure. The truth is that calling "itself" via a grpc client is mainly for dev/testing purposes but it would not hurt to keep the ports in sync, or?

**Comment by Sgitario** [comment]
> I'm not sure. The truth is that calling "itself" via a grpc client is mainly for dev/testing purposes but it would not hurt to keep the ports in sync, or?

+100

**Comment by michalszynkiewicz** [comment]
For testing one can use a client without configuration: https://quarkus.io/guides/grpc-service-implementation#testing-your-services

I'm not sure I understand the dev purposes of exposing a client inside the same application. Could you elaborate?

**Comment by michalszynkiewicz** [comment]
What I think we could do better is setting host and port (if unset) to the exposed ones for tests, even for the SSL ones (ATM we do that only for plaintext ones).

**Comment by Sgitario** [comment]
> > For testing one can use a client without configuration: https://quarkus.io/guides/grpc-service-implementation#testing-your-services
> > I'm not sure I understand the dev purposes of exposing a client inside the same application. Could you elaborate?
> > TBH I find the fact that port is optional a bit strange. We don't do default ports for http clients.
> > I'm slightly against this change because it will break users who use a non-default port to expose services and rely on us setting the default to 9000 to talk with an external service.
> > Even if for demos we often do a client that talks with our own app, I don't think that's what the normal users do.
> 
> I don't see the impact you state here as for users that do not use the default port 9000 for the client, they need to set the clients property anyway, so no difference here.
> 
> Let me rephrase this change: if users are using the same port as our server's default port, you don't need to change anything which leads to a better user experience.

Anyway, @michalszynkiewicz if you still think this is not a plus, let me know and I will close this PR and the issue as it's not that important.

**Comment by gsmet** [comment]
I tend to agree with @michalszynkiewicz . I don't think changing the server port should change the default client one given your clients will probably talk to external independent servers.

**Comment by Sgitario** [comment]
> I tend to agree with @michalszynkiewicz . I don't think changing the server port should change the default client one given your clients will probably talk to external independent servers.

Okis, closing!
